### PR TITLE
feat: implement DuckDBConnection#getSchema for JDBC

### PR DESCRIPTION
**Pull request by Mause**
First time in a long time that I've written any C++ or used CMake 😅

Any pointers (😉) are very welcome!

### CODE REVIEW

File: tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java [hunk 1]
```diff
@@ -2159,14 +2160,36 @@ public static void test_unsiged_integers() throws Exception {
 		conn.close();
 	}
 
+
+	public static void test_get_schema() throws Exception {
+		DuckDBConnection conn = (DuckDBConnection) DriverManager.getConnection("jdbc:duckdb:");
+
+		assertEquals(conn.getSchema(), "main");
```
Comment (Mytherin) [inline tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java:2167]: Could we also add a test where we change the schema, and verify that `getSchema` correctly returns the modified schema?

File: tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java [hunk 2]
```diff
@@ -2159,14 +2160,36 @@ public static void test_unsiged_integers() throws Exception {
...
 	public static void main(String[] args) throws Exception {
 		// Woo I can do reflection too, take this, JUnit!
 		Method[] methods = TestDuckDBJDBC.class.getMethods();
+		boolean anyFailed = false;
 		for (Method m : methods) {
 			if (m.getName().startsWith("test_")) {
 				m.invoke(null);
+				System.out.print(m.getName() + " ");
+
+				LocalDateTime start = LocalDateTime.now();
```
Comment (Mytherin) [inline tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java:2180]: This looks like debug code - should this be merged?

**Review by Mytherin — DISMISSED** [review]
Thanks for the PR! Looks great. Some comments:

File: tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java (hunk 2 above)
Comment (Mause) [reply tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java:2192]: I can remove the timing code, I mostly added it to show how slow some of these tests were (most are less than a second, two are over a minute each). The rest I think is useful, as long as we aren't using a real test runner such as junit or testng

File: tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java (hunk 1 above)
Comment (Mause) [reply tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java:2170]: Done!

File: tools/jdbc/src/jni/duckdb_java.cpp [hunk 3]
```diff
@@ -230,6 +232,19 @@ JNIEXPORT jobject JNICALL Java_org_duckdb_DuckDBNative_duckdb_1jdbc_1connect(JNI
 	return nullptr;
 }
 
+JNIEXPORT jstring JNICALL Java_org_duckdb_DuckDBNative_duckdb_1jdbc_1get_1schema(JNIEnv *env, jclass,
+                                                                                 jobject conn_ref_buf) {
+	auto conn_ref = (Connection *)env->GetDirectBufferAddress(conn_ref_buf);
+	if (!conn_ref || !conn_ref->context) {
+		env->ThrowNew(J_SQLException, "Invalid connection");
```
[the remark this answers is not in the document]
Comment (Mause) [reply tools/jdbc/src/jni/duckdb_java.cpp]: Done!

**Comment by Mytherin** [comment]
Looks good, thanks!

**Pull request merged**